core/src/main/java/org/opensearch/sql/expression/Expression.java

```diff
@@ -19,8 +20,11 @@ public interface Expression extends Serializable {
   /**
    * Evaluate the value of expression in the value environment.
    */
-  ExprValue valueOf(Environment<Expression, ExprValue> valueEnv);
+  ExprValue valueOf(Environment<Expression, ExprValue> valueEnv, SessionContext sessionContext);
```

Comment:
Does it make sense to provide an overloaded version of `valueOf` that takes just an `Environment`? 
```
ExprValue valueOf(Environment<Expression, ExprValue> valueEnv) {
    return valueOf(valueEnv, SessionContext.None);
  }
```
This would cut down the number of changes needed in this PR